cmd/kubeadm/app/cmd/phases/addons.go

```diff
@@ -17,186 +17,110 @@ limitations under the License.
-	"github.com/spf13/cobra"
+	"github.com/pkg/errors"
 
 	clientset "k8s.io/client-go/kubernetes"
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
-	kubeadmscheme "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme"
-	kubeadmapiv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
-	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation"
 	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
 	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
-	"k8s.io/kubernetes/cmd/kubeadm/app/features"
 	dnsaddon "k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns"
 	proxyaddon "k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy"
-	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
-	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
-	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 	"k8s.io/kubernetes/pkg/util/normalizer"
 )
 
 var (
-	allAddonsLongDesc = normalizer.LongDesc(`
-		Installs the CoreDNS and the kube-proxy addons components via the API server.
-		Please note that although the DNS server is deployed, it will not be scheduled until CNI is installed.
-		` + cmdutil.AlphaDisclaimer)
-
-	allAddonsExample = normalizer.Examples(`
-		# Installs the CoreDNS and the kube-proxy addons components via the API server,
-		# functionally equivalent to what installed by kubeadm init. 
-
-		kubeadm alpha phase self-hosting from-staticpods
-		`)
-
 	corednsAddonsLongDesc = normalizer.LongDesc(`
 		Installs the CoreDNS addon components via the API server.
 		Please note that although the DNS server is deployed, it will not be scheduled until CNI is installed.
-		` + cmdutil.AlphaDisclaimer)
+		`)
 
 	kubeproxyAddonsLongDesc = normalizer.LongDesc(`
 		Installs the kube-proxy addon components via the API server.
-		` + cmdutil.AlphaDisclaimer)
+		`)
 )
 
-// NewCmdAddon returns the addon Cobra command
-func NewCmdAddon() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:     "addon",
-		Aliases: []string{"addons"},
-		Short:   "Installs required addons for passing Conformance tests",
-		Long:    cmdutil.MacroCommandLongDescription,
-	}
-
-	cmd.AddCommand(getAddonsSubCommands()...)
-	return cmd
+type addonData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	Client() (clientset.Interface, error)
 }
 
-// EnsureAllAddons installs all addons to a Kubernetes cluster
-func EnsureAllAddons(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error {
-
-	addonActions := []func(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error{
-		dnsaddon.EnsureDNSAddon,
-		proxyaddon.EnsureProxyAddon,
+// NewAddonPhase returns the addon Cobra command
+func NewAddonPhase() workflow.Phase {
+	return workflow.Phase{
+		Name:     "addon",
+		Short:    "Installs required addons for passing Conformance tests",
+		Long:     cmdutil.MacroCommandLongDescription,
```

Comment:
we should remove Long.
it would contain:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/util/documentation.go#L31

it's not true because the 'addon' command can be called on it's own (calls both sub-phases).